packages/flutter/lib/src/material/stepper.dart

```diff
@@ -0,0 +1,577 @@
+    return new SizedBox(
+      width: visible ? 1.0 : 0.0,
+      height: 16.0,
+      child: new Container(
+        decoration: new BoxDecoration(
+          backgroundColor: Colors.grey[400]
+        )
+      )
+    );
+  }
+
+  Widget _buildCircle(BuildContext context, int index) {
+    Widget container;
+
+    if (config.steps[index].state != StepState.error) {
+      container = new AnimatedContainer(
+        curve: Curves.fastOutSlowIn,
+        duration: kThemeAnimationDuration,
+        decoration: new BoxDecoration(
+          backgroundColor: config.steps[index].state == StepState.active ?
+              Theme.of(context).primaryColor : Colors.black38,
+          shape: BoxShape.circle
+        ),
+        child: new Center(
+          child: config.steps[index].circleChild ??
+              new Text('${index + 1}', style: _kStepStyle)
+        )
+      );
+    } else {
+      container = new Center(
+        child: new SizedBox(
+          width: 24.0,
+          height: 20.78, // Height of 24dp-long-sided equilateral triangle.
+          child: new CustomPaint(
+            painter: new _TrianglePainter(
+              color: Colors.red[500]
+            ),
+            child: new Align(
+              alignment: const FractionalOffset(0.5, 0.9), // 0.9 looks better than the geometrical 0.66.
+              child: config.steps[index].circleChild ??
+                  new Text('!', style: _kStepStyle)
+            )
+          )
+        )
+      );
+    }
+
+    return new Container(
+      margin: const EdgeInsets.symmetric(vertical: 8.0),
+      child: new SizedBox(
+        width: 24.0,
+        height: 24.0,
+        child: container
+      )
+    );
+  }
+
+  Widget _buildVerticalControls() {
+    return new Container(
+      margin: const EdgeInsets.only(top: 16.0),
+      child: new ConstrainedBox(
+        constraints: const BoxConstraints.tightFor(height: 48.0),
+        child: new Row(
+          children: <Widget>[
+            new FlatButton(
+              onPressed: config.onStepContinue,
+              color: Theme.of(context).primaryColor,
+              textColor: Colors.white,
+              textTheme: ButtonTextTheme.normal,
+              child: new Text('CONTINUE')
+            ),
+            new Container(
+              margin: const EdgeInsets.only(left: 8.0),
+              child: new FlatButton(
+                onPressed: config.onStepCancel,
+                textColor: Colors.black54,
+                textTheme: ButtonTextTheme.normal,
+                child: new Text('CANCEL')
+              )
+            )
+          ]
+        )
+      )
+    );
+  }
+
+  TextStyle _titleStyle(int index) {
+    switch (config.steps[index].state) {
+      case StepState.disabled:
+        return _kDisabledTitleStyle;
+      case StepState.active:
+      case StepState.inactive:
+        return _kActiveTitleStyle;
+      case StepState.error:
+        return _kErrorTitleStyle;
+      default:
```

Comment:
remove the default: from switches on enums, otherwise the analyzer won't catch when you omit a value